### PR TITLE
2017-01-13-d3-brush

### LINKED ISSUE #28: 2017-01-13 - Jeffrey - Topic: tbd

### PR DESCRIPTION
**Pull request by Jephuff**
closes #28

### CODE REVIEW

File: _posts/jeffrey_burt/2017-01-13-d3-brush.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+```javascript
+var scaleLinear = d3.scaleLinear || d3.scale.linear;
+```
+
+and then call `scaleLinear` where ever it's needed.
+
+## Using the brush
+The biggest change we had to make was where we were using the brush for zooming  
+
+<img src="/images/jeffrey_burt/brush-selection.gif" alt="brush-selection" style="max-width: 320px;"/>
```
Comment (shanewilson) [inline _posts/jeffrey_burt/2017-01-13-d3-brush.md:52]: should this be an absolute path?

File: _posts/jeffrey_burt/2017-01-13-d3-brush.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    </tr>
+    <tr>
+        <td>wrapper.select('.brush').call(brush.clear())</td>
+        <td>wrapper.select('.brush').call(brush.move, null)</td>
+    </tr>
+</table>
+
+Because v4 removed the `brush.clear` method, we had to replaced it with a `brush.move` call to `null`  
+
+This introduces another issue. The `brush.clear` method would not fire the `brushend` event, which meant it was fine to call it inside the `brushend` handler itself. Now that we were using the `brush.move` method, it would fire the `end` event and therefore cause a loop. To solve this, we heck for `d3.event.selection`, if there is nothing there, return.
```
Comment (shanewilson) [inline _posts/jeffrey_burt/2017-01-13-d3-brush.md]: we heck for `d3.event.selection`,

**Review by shanewilson — CHANGES_REQUESTED** [review]

File: _posts/jeffrey_burt/2017-01-13-d3-brush.md (hunk 2 above)
Comment (shanewilson) [reply _posts/jeffrey_burt/2017-01-13-d3-brush.md]: also maybe too many commas in that last sentence.

File: _posts/jeffrey_burt/2017-01-13-d3-brush.md (hunk 1 above)
Comment (Jephuff) [reply _posts/jeffrey_burt/2017-01-13-d3-brush.md:52]: seemed better then `../../../../../images/jeffrey_burt/brush-selection.gif` haha

Comment (shanewilson) [reply _posts/jeffrey_burt/2017-01-13-d3-brush.md:52]: oh I thought that was a local path

Comment (Jephuff) [reply _posts/jeffrey_burt/2017-01-13-d3-brush.md:52]: ahh, yeah It's pointing to the root images directory.

**Review by shanewilson — APPROVED** [review]

**Pull request merged**